mysql-test/t/percona_slowlog_size_limits.test

```diff
@@ -0,0 +1,237 @@
+#
+# Test slowlog size limiting and rotation
+#
+
+SET @old_slow_query_log = @@global.slow_query_log;
+SET @old_log_output = @@global.log_output;
+SET @old_slow_query_log_file = @@global.slow_query_log_file;
+SET @old_max_slowlog_size = @@global.max_slowlog_size;
+SET @old_slowlog_space_limit = @@global.slowlog_space_limit;
+SET @old_long_query_time = @@long_query_time;
+SET long_query_time = 0;
+SET GLOBAL log_output = FILE;
+SET GLOBAL slow_query_log = 1;
+
+
+--echo # Case 1 (PS-1484): Test if slowlog_space_limit is working correctly
+--echo #                   with the default slow query log name
+SET GLOBAL max_slowlog_size = 4096;
+SET GLOBAL slowlog_space_limit = @@global.max_slowlog_size * 3;
+
+# avoid 100 selects 1 in .result
+--let $i=100
+--disable_query_log
+--disable_result_log
+while($i) {
+  --eval select $i
+  --dec $i
+}
+--enable_query_log
+--enable_result_log
+--let SLOW_QUERY_BASE= `SELECT @old_slow_query_log_file`
+--echo slow query log files matching 'mysqld-slow.log.*' count:
+--perl
+my @files = <$ENV{'SLOW_QUERY_BASE'}.*>;
+printf "%d\n", scalar(@files);
```

Comment:
May be write this `scalar(@files)` to a temp file (something like `$MYSQL_TMP_DIR/list_result.inc`) inside this `--perl` block in format
```perl
printf FILEHANDLE "--let \$number_of_files = %d\n", scalar(@files);`
```
and then include this file in MTR immediately after the `--perl` block
```
--source $MYSQL_TMP_DIR/list_result.inc
```
This way you will be able to assert `$number_of_files == nnn`.

At the end, do not forget to remote this `$MYSQL_TMP_DIR/list_result.inc`